Common/src/Common/Service/Data/Search/Search.php

```diff
@@ -342,11 +342,10 @@ public function updateFilterValuesFromForm(): void
         );
 
         foreach ($this->getFilters() as $filterClass) {
-            /** @var \Common\Data\Object\Search\Aggregations\Terms\TermsAbstract $filterClass */
             if (!isset($post['filter'][$filterClass->getKey()])) {
                 continue;
             }
-            if (empty($post['filter'][$filterClass->getKey()])) {
+            if (empty($post['filter'][$filterClass->getKey()]) && $post['filter'][$filterClass->getKey()] !== '0') {
```

Comment:
Original logic was: `$post['filter'][$filterClass->getKey()] !== ''` may be better to revert back to this instead.

Can we look to get test coverage on this part of the code please so we don't regress again?

Reply:
Yeah that was what I did originally but it got changed while I was away, to reduce stan errors so a ticket I had ready for test failed. Level must have been changed in the interim maybe?

Reply:
Yeah, regressed in: https://github.com/dvsa/olcs-common/commit/3ad9d2d10caab8e890a7a95e06c4aba44d256a50.

Reply:
Reverted and test added to prevent regression